qutip/qobj.py

```diff
@@ -1234,48 +1230,50 @@ def sqrtm(self, sparse=False, tol=0, maxiter=100000):
             raise TypeError('Invalid operand for matrix square root')
 
     def cosm(self):
-        """Cosine of a quantum operator.
+        """Cosine of a square quantum operator.
 
         Operator must be square.
 
         Returns
         -------
-        oper : :class:`qutip.Qobj`
-            Matrix cosine of operator.
+        :obj:`.Qobj`
+            Matrix cosine of an operator.
 
         Raises
         ------
-        TypeError
-            Quantum object is not square.
+        TypeError : Invalid operand for matrix square root.
+            When input is not a square operator, number of rows and columns
+            do not match.
 
         Notes
         -----
-        Uses the Q.expm() method.
+        Uses the Q. :obj:`.expm()` method.
 
         """
         if self.dims[0][0] == self.dims[1][0]:
             return 0.5 * ((1j * self).expm() + (-1j * self).expm())
         else:
-            raise TypeError('Invalid operand for matrix square root')
+            raise TypeError('Invalid operand for matrix square root.')
 
     def sinm(self):
-        """Sine of a quantum operator.
+        """Sine of a square quantum operator.
 
         Operator must be square.
 
         Returns
         -------
-        oper : :class:`qutip.Qobj`
+        :class:`Qobj`
             Matrix sine of operator.
 
         Raises
         ------
-        TypeError
-            Quantum object is not square.
+        TypeError : Invalid operand for matrix square root
+            When input is not a square operator, number of rows and columns
+            do not match.
 
         Notes
         -----
-        Uses the Q.expm() method.
+        Uses the Q. :obj:`.expm()` method.
```

Comment:
this looks very weird in the rendered version.  You can just say "Uses the `` :obj:`~.expm` `` method" - it's clear from context that it refers to `Qobj`, and avoids the weird split between the Q and the monospace font.